parsec/mca/pins/task_profiler/pins_task_profiler_module.c

```diff
@@ -97,6 +174,11 @@ static void pins_fini_task_profiler(parsec_context_t *master_context)
 static void pins_thread_init_task_profiler(struct parsec_execution_stream_s * es)
 {
     parsec_pins_next_callback_t* event_cb;
+    event_cb = (parsec_pins_next_callback_t*)malloc(sizeof(parsec_pins_next_callback_t));
```

Comment:
What is the point of registering events that will not be generated ?

Reply:
Are you referring to the PREPARE_INPUT events?

Reply:
We now have a mask of events that will be generated. If we move the reading the of MCA earlier in this function, we can only register the events that are useful.

Reply:
Ah, I see—so rather than registering callbacks for all events, but those events never being generated due to the `parsec_pins_enable_mask`, simply don't register callbacks for those events.

This also lets a different PINS module listen to those events if it wants. The current solution has the MCA param "belong" to the task_profiler module, but it changes what PINS events are generated for _all_ PINS modules, not just itself.